lib/fluent/plugin/out_collectd_nest.rb

```diff
@@ -39,16 +39,26 @@ def normalize_record(record)
       if !(record.has_key?('values')) || !(record.has_key?('dsnames')) || !(record.has_key?('dstypes')) || !(record.has_key?('host')) || !(record.has_key?('plugin')) || !(record.has_key?('type'))
         return record
       end
-      new_rec = {}
-      new_rec['hostname']= record['host']
-      rec_plugin = record['plugin']
-      rec_type = record['type']
-      record[rec_plugin] = {rec_type => {}}
-
-      record['values'].each_with_index { |value, index|
-        record[rec_plugin][rec_type][record['dsnames'][index]] = value
-      }
-      record['dstypes'] = record['dstypes'].uniq
+      if record['dsnames'].length == 1
+          new_rec = {}
+          new_rec['hostname']= record['host']
+          rec_plugin = record['plugin']
+          rec_type = record['type']
+          record[rec_plugin] = {rec_type => {}}
+
```

Comment:
I think there is a miscommunication here.
the value of `dsnames` field is essentially lost here.